common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -110,7 +110,9 @@ private String getStacktrace(Throwable e) {
 
     public void close() {
         try {
-            dataPublisher.shutdown();
+            if(dataPublisher!=null){
```

Comment:
This is fixed in following commit :  https://github.com/wso2/shared-analytics/pull/21/commits/eb6f50ab69a03c7b98f0a9aca51cc43b9139b46e